app/persistence/spread_repository.py

```diff
@@ -0,0 +1,165 @@
+import psycopg2
+import os
+from dotenv import load_dotenv, find_dotenv
+
+load_dotenv(find_dotenv())
+
+
+def connect_db():
+    """
+    Connect to the database.
+
+    :return: Connection to the  database.
+    """
+    database_name = os.getenv("DATABASE_NAME")
+    database_host = os.getenv("DATABASE_HOST")
+    database_port = os.getenv("DATABASE_PORT")
+    database_user = os.getenv("DATABASE_USER")
+    database_password = os.getenv("DATABASE_PASSWORD")
+
+    conn = psycopg2.connect(
+        database=database_name,
+        host=database_host,
+        user=database_user,
+        password=database_password,
+        port=database_port,
+    )
+    return conn
+
+
+def get_spreads():
+    """
+    Get all the spread in the database.
+
+    :return: A list of all spread.
+    """
+    conn = connect_db()
+    cur = conn.cursor()
+    cur.execute("SELECT * FROM spread")
+    spreads_data = cur.fetchall()
+    spreads_list = []
+    for spread in spreads_data:
+        spread_dict = {
+            "fertilizer_id": spread[0],
+            "plot_number": spread[1],
+            "date": spread[2],
+            "spread_quantity": float(spread[3]),
+        }
+        spreads_list.append(spread_dict)
+    conn.close()
+    return spreads_list
+
+
+def get_spread_by_fertilizer(fertilizer_id):
+    """
+    Get a spread by fertilizer_id.
+
+    :param (str) fertilizer_id: The ID of the fertilizer.
+    :return: A list of dict with the spread's data or None if not found.
+    """
+    conn = connect_db()
+    cur = conn.cursor()
+    cur.execute(
+        "SELECT * FROM spread WHERE fertilizer_id = %s",
+        (fertilizer_id,),
+    )
+    spread = cur.fetchone()
+    conn.close()
+
+    if spread is None:
+        return None
+
+    spread_dict = {
+        "fertilizer_id": spread[0],
+        "plot_number": spread[1],
+        "date": spread[2],
+        "spread_quantity": float(spread[3]),
+    }
+
+    return spread_dict
+
+
+def create_spread(fertilizer_id, plot_number, date, spread_quantity):
+    """
+    Create a new spread in the database.
+
+    :param (str) fertilizer_id: The ID of the fertilizer.
+    :param (int) plot_number: The plot number.
+    :param (date) date: The date of the spread.
+    :param (int) spread_quantity: The quantity of the spread.
+    """
+    existing_spread = get_spread_by_fertilizer(fertilizer_id)
```

Comment:
This is done in the route side.

Reply:
I need it here so I don't get an internal server error.